research/minigo/model_utils.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2018 MLBenchmark Group. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities for DualNet model."""
+
+# pylint: disable=g-bad-import-order
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy
+import os
+import petname
+import re
+
+import tensorflow as tf
+
+from tensorflow.python.training.summary_io import SummaryWriterCache
+
+# Regular expression of model number and name.
```

Comment:
Comments! What format is this capturing? Remember, one of my goals is to never have to manually parse regex strings.